src/views/ChangePassword/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { Button, Stack, Typography } from '@mui/material';
+import { Box } from '@mui/system';
+import { useTranslation } from 'react-i18next';
+import PasswordTextfield from 'components/PasswordTextfield';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import CancelButton from 'components/CancelButton';
+
+interface ChangePasswordForm {
+  currentPassword: string;
+  newPassword: string;
+}
+
+const changePasswordSchema = yup
+  .object({
+    currentPassword: yup.string().min(8, 'Must be minimum ${min} characters').required('Field is required'),
+    newPassword: yup
+      .string()
+      .min(8, 'Must be minimum ${min} characters')
+      .required('Field is required')
+      .notOneOf([yup.ref('currentPassword'), null], 'Passwords must be different'),
+  })
+  .required();
```

Comment:
Why are you using `yup` when react-hook-form has its own solution?
https://react-hook-form.com/get-started/#Applyvalidation

Reply:
Sorry, I am silly and instead of looking for any solutions that use native `react-hook-form` validation, I assumed that it's impossible to do without `yup`.  😄